modules/tinymce/src/themes/silver/main/ts/ui/dialog/UrlInput.ts

```diff
@@ -236,7 +236,7 @@ export const renderUrlInput = (
   const memUrlPickerButton = Memento.record(renderButton({
     name: spec.name,
     icon: Optional.some('browse'),
-    text: spec.label.getOr(''),
+    text: spec.picker_text.getOr(spec.label.getOr('')),
```

Comment:
This might be slightly cleaner
```suggestion
    text: spec.picker_text.or(spec.label).getOr(''),
```

Reply:
Agreed 👍